runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -54,17 +54,11 @@ public class LogReplicationUtils {
     private static final int DEFAULT_BUFFER_SIZE = -1;
 
     public LogReplicationUtils(CorfuStore corfuStore) {
-        try {
-            this.corfuStore = corfuStore;
-            corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, REPLICATION_STATUS_TABLE, ReplicationStatusKey.class,
-                    ReplicationStatusVal.class, null, TableOptions.fromProtoSchema(ReplicationStatusVal.class));
-            schedulerThread = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
-                    .setNameFormat(LogReplicationUtils.class.getName())
-                    .build());
-        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
-            log.error("Failed to open the replication status table", e);
-            throw new UnrecoverableCorfuError(e);
-        }
+        this.corfuStore = corfuStore;
+
+        schedulerThread = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
```

Comment:
this is because I wanted to be sure that the task ran immediately on subscription.  Without the scheduled executor service, there is no guarantee when the task will run.

One open item now is shutting down the executor service.  There is no such method available in corfuStore.  Should I add one?

Reply:
good point, definitely we need to shut it down. Add which method? - right after we complete the subscription you could shut it down.

Reply:
actually, we shouldnt shut it down after completing the subscription because the client may need to resubscribe if they hit a trimmed exception or other error through the onError() callback

I am shutting it down alongwith the runtime shutdown.

We can discuss further if you think I am missing something.